src/main/java/practice/StreamPractice.java

```diff
@@ -14,16 +18,26 @@ public class StreamPractice {
      * "Can't get min value from list: < Here is our input 'numbers' >"
      */
     public int findMinEvenNumber(List<String> numbers) {
-        return 0;
+        return numbers.stream()
+                .flatMapToInt(num -> Arrays.stream(num.split(","))
+                        .mapToInt(Integer::parseInt))
+                .filter(num -> num % 2 == 0)
+                .min()
+                .orElseThrow(() -> new RuntimeException("Can't get min value from list:"
+                        + numbers));
     }
 
     /**
      * Given a List of Integer numbers,
      * return the average of all odd numbers from the list or throw NoSuchElementException.
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
-    public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+    public double getOddNumsAverage(List<Integer> numbers) {
+        return IntStream.range(0, numbers.size())
+                .map(i -> i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
+                .filter(n -> n % 2 != 0)
+                .average()
+                .orElseThrow();
```

Comment:
use getAsDouble() method
```suggestion
                .getAsDouble();
```